crates/common/src/add_order.rs

```diff
@@ -70,7 +70,7 @@ impl AddOrderArgs {
             if let Some(decimals) = input.token.decimals {
                 inputs.push(IO {
                     token: input.token.address,
-                    vaultId: input.vault_id,
+                    vaultId: input.vault_id.unwrap_or(random_u256()),
```

Comment:
isn't this going to produce a different vault id for every input and output?

that seems like it's obviously not what people want/expect